include/deal.II/lac/block_vector_base.h

```diff
@@ -1514,6 +1522,16 @@ BlockVectorBase<VectorType>::size () const
 
 template <class VectorType>
 inline
+std::size_t
+BlockVectorBase<VectorType>::local_size () const
+{
+  return size();
+}
```

Comment:
I don't particularly like having the `local_size` method in the block vector implemented this way because we would need to change that method in derived classes whose local size is actually different from the size.

The correct thing to do would be to compute the local size as a sum of the local sizes on each participating vector. Since there is no useful access to the local indices in a block vector (at least as far as I can tell), I would directly compute that sum here, even though it would make for loops that put `i<local_size()` quite expensive...